crates/wit-component/src/encoding/wit.rs

```diff
@@ -57,11 +157,44 @@ impl Encoder<'_> {
         // decoding process where everyone's view of a foreign document agrees
         // notably on the order that types are defined in to assist with
         // roundtripping.
-        let mut interfaces = IndexSet::new();
-        for (_, id) in self.resolve.packages[self.package].interfaces.iter() {
-            self.add_live_interfaces(&mut interfaces, *id);
+        for (name, &id) in self.resolve.packages[self.package].interfaces.iter() {
```

Comment:
Should the comment above this code be reworded given these changes?  It seems odd that the exact same comment is duplicated below in `encode_interface`; it seems like a better fit there than here.